tree-sitter/src/TreeSitter/Symbol.hs

```diff
@@ -114,11 +114,12 @@ escapeOperatorPunctuation = concatMap $ \case
 
 -- | Convert a snake_case String to camelCase
 camelCase :: String -> String
-camelCase = foldr appender mempty
+camelCase = go
   where
-    appender :: Char -> String -> String
-    appender '_' cs = capitalize cs
-    appender c cs   = c : cs
+    go ('_':'_':xs) = "Underscore" <> go xs
+    go ('_':xs)     = go (capitalize xs)
+    go (x:xs)       = x : go xs
+    go ""           = ""
```

Comment:
Will this require any kind of version bump?

Reply:
We'll have to release new versions of tree-sitter and tree-sitter-$lang to get all this stuff out. I don't think this is a break change as we hadn't yet tried to generate syntax for things like `__END__` in Ruby/Pearl and the existing generated syntax for Python/Java won't change (it's only double underscore that has to be different than just a single underscore).